test/PushNotificationsTests/APITests.cpp

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include <testdef.h>
+#include "Shared.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt;
+using namespace winrt::Microsoft::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel;
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+
+// TODO: Write Register/Unregister tests that utilize the Assoc APIs to validate results.
+
+namespace Test::PushNotifications
+{
+    class APITests
+    {
+    private:
+        wil::unique_event m_failed;
+
+        const std::wstring c_testDataFileName = L"testfile" + c_testFileExtension;
+        const std::wstring c_testDataFileName_Packaged = L"testfile" + c_testFileExtension_Packaged;
+        const std::wstring c_testPackageFile = g_deploymentDir + L"AppLifecycleTestPackage.msixbundle";
+        const std::wstring c_testPackageCertFile = g_deploymentDir + L"AppLifecycleTestPackage.cer";
+        const std::wstring c_testPackageFullName = L"AppLifecycleTestPackage_1.0.0.0_x64__ph1m9x8skttmg";
+
+    public:
+        BEGIN_TEST_CLASS(APITests)
+            TEST_CLASS_PROPERTY(L"IsolationLevel", L"Method")
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunFixtureAs:Class", L"RestrictedUser")
+            END_TEST_CLASS()
+
+            TEST_CLASS_SETUP(ClassInit)
+        {
+            ::Test::Bootstrap::SetupPackages();
+
+            // Write out some test content.
+            WriteContentFile(c_testDataFileName);
+            WriteContentFile(c_testDataFileName_Packaged);
+
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            // Swallow errors in cleanup.
+            try
+            {
+                DeleteContentFile(c_testDataFileName_Packaged);
+                DeleteContentFile(c_testDataFileName);
+                //UninstallPackage(c_testPackageFullName);
+            }
+            catch (const std::exception&)
+            {
+            }
```

Comment:
I think that the test should log a message at the very least if we fail certain parts of the cleanup operation. What would be the repercusion of failing deletion above? Would the follow up tests fail if we re-run the tests?

Reply:
And if the result puts us in a bad state, we probably want to fail the test itself?

Reply:
In the future, we will want to use InstallPackage/RunCertUtil and in the catch return false if those calls fail. We won't use Write/DeleteContentFile I don't think.